virtio-devices/src/iommu.rs

```diff
@@ -407,13 +407,35 @@ impl Request {
                     let bypass =
                         (req.flags & VIRTIO_IOMMU_ATTACH_F_BYPASS) == VIRTIO_IOMMU_ATTACH_F_BYPASS;
 
+                    let mut old_domain_id = domain_id;
+                    if let Some(&id) = mapping.endpoints.read().unwrap().get(&endpoint) {
+                        old_domain_id = id;
+                    }
+
+                    if old_domain_id != domain_id {
+                        detach_endpoint_from_domain(endpoint, old_domain_id, mapping, ext_mapping)?;
+                    }
+
                     // Add endpoint associated with specific domain
                     mapping
                         .endpoints
                         .write()
                         .unwrap()
                         .insert(endpoint, domain_id);
 
+                    // If any other mappings exist in the domain for other containers,
```

Comment:
This is also a separate bug-fix than what is specified in the PR description and commit message. Can you either update the commit message or split this change into a separate commit.

Support for adding an endpoint to an existing domain looks like optional behavior. The specification also allows the device to return `VIRTIO_IOMMU_S_UNSUPP`. This is a good fix because it looks like we were failing silently before.

Reply:
Commit message has been updated - can split into an additional PR but don't want to miss it and it's still required for nested virtualization reboot to work.

Reply:
Follow-up split this into a second commit in this PR